koschei/backend/services/repo_resolver.py

```diff
@@ -56,7 +57,16 @@ class RepoGenerationException(Exception):
 class RepoResolver(Resolver):
     def main(self):
         for collection in self.db.query(Collection).all():
-            self.process_repo(collection)
+            try:
+                pg_lock(
+                    self.db, LOCK_REPO_RESOLVER, collection.id,
+                    block=False, transaction=True,
+                )
+                self.process_repo(collection)
```

Comment:
You are using trancaction-bound lock, but process_repo() commits multiple transactions.

Reply:
Good point. Even though other resolver processes won't pick the collection for processing of the same repo (the last repo_id is stored in the first transaction), it could pick a new one generated in the meantime and stomp on each other. Will fix.